src/EfficientDynamoDb/Configs/RegionEndpoint.cs

```diff
@@ -10,136 +10,141 @@ public class RegionEndpoint
         internal string RequestUri { get; }
         
         public string Region { get; }
+
+        static RegionEndpoint Create(string region, string uriFormat)
+        {
+            return new RegionEndpoint(region, string.Format(uriFormat, ServiceName, region));
+        }
         
-        private RegionEndpoint(string region, string uriFormat)
+        public RegionEndpoint(string region, string requestUri)
```

Comment:
I'm not a fan of creating a public constructor in this case because of the following:
1. People usually try to create an object using constructors first. So when it's present, they might not find static factory-like properties but pass region strings to the constructor.
2. There is no straightforward way to reuse existing region factories when connecting to localdb. Users would need to pass a region string directly here or use an ugly construction:
```csharp
var region = new RegionEndpoint(RegionEndpoint.USEast1.Region, "http://localhost:8000");
// or
var region = new RegionEndpoint("us-east-1", "http://localhost:8000");
```
What do you think about adding an instance method `WithUrl(string url)` that will create a new instance of `RegionEndpoint` with an old region but a new URL? In this way, we can reuse all existing factory methods. E.g.:
```csharp
var globalUsEast = RegionEndpoint.USEast1;
var localUsEast = RegionEndpoint.USEast1.WithUrl("http://localhost:8000");
```

That's true that adding a new region will require a library update. To handle that case I'd prefer using some public static factory method with one `region` parameter instead of a constructor here. Like this: `RegionEndpoint.Create("new_region")`

Reply:
I agree with your point about the constructor being public, people will look to use that first. What if we left that alone, and instead had the Create static method do this custom construction? That way, when users don't see a constructor they'll likely discover both the Create method and the pre-defined region instances.

Reply:
That sounds good. But I believe we need two overloads:
```csharp
public static RegionEndpoint Create(string region);
public static RegionEndpoint Create(string region, string endpointUrl);
```
The first one is going to be used when you need to create a newly added AWS region. And the second one will be used when you need to connect to a local db.

The only thing I don't like here is that with these 2 overloads, creating a region in local db is not really elegant. What if we add a third overload that accepts `RegionEndpoint region` parameter instead of `string region`? So the usage will look like this:
```csharp
var localRegion = RegionEndpoint.Create(RegionEndpoint.USEast1, "http://localhost:8000");
```